cucumber-java8/src/main/java/io/cucumber/java8/ClosureAwareGlueRegistry.java

```diff
@@ -0,0 +1,113 @@
+package io.cucumber.java8;
+
+import io.cucumber.core.backend.DataTableTypeDefinition;
+import io.cucumber.core.backend.DefaultDataTableCellTransformerDefinition;
+import io.cucumber.core.backend.DefaultDataTableEntryTransformerDefinition;
+import io.cucumber.core.backend.DefaultParameterTransformerDefinition;
+import io.cucumber.core.backend.DocStringTypeDefinition;
+import io.cucumber.core.backend.Glue;
+import io.cucumber.core.backend.HookDefinition;
+import io.cucumber.core.backend.ParameterTypeDefinition;
+import io.cucumber.core.backend.StepDefinition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+final class ClosureAwareGlueRegistry implements LambdaGlueRegistry {
```

Comment:
This class needs a sanity check to ensure that the number of abstract glue definitions on the first run matches the number or registered/updated glue definitions on any subsequent runs.


```
if(registered != definitions.size()) {
   throw new CucumberException...
}
```

Could be invokes at the end of `buildWorld`?